homeassistant/components/reolink/__init__.py

```diff
@@ -0,0 +1,98 @@
+"""Reolink integration for HomeAssistant."""
+
+from __future__ import annotations
+
+import asyncio
+from dataclasses import dataclass
+from datetime import timedelta
+import logging
+
+from aiohttp import ClientConnectorError
+import async_timeout
+from reolink_ip.exceptions import ApiError, InvalidContentTypeError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DEVICE_UPDATE_INTERVAL, DOMAIN, PLATFORMS
+from .host import ReolinkHost
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class ReolinkData:
+    """Data for the Reolink integration."""
+
+    host: ReolinkHost
+    device_coordinator: DataUpdateCoordinator
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Reolink from a config entry."""
+    host = ReolinkHost(hass, dict(entry.data), dict(entry.options))
+
+    try:
+        if not await host.async_init():
+            raise ConfigEntryNotReady(
+                f"Error while trying to setup {host.api.host}:{host.api.port}: failed to obtain data from device."
+            )
+    except (
+        ClientConnectorError,
+        asyncio.TimeoutError,
+        ApiError,
+        InvalidContentTypeError,
+    ) as err:
+        raise ConfigEntryNotReady(
+            f'Error while trying to setup {host.api.host}:{host.api.port}: "{str(err)}".'
+        ) from err
+
+    entry.async_on_unload(
+        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, host.stop)
+    )
+
+    async def async_device_config_update():
+        """Perform the update of the host config-state cache, and renew the ONVIF-subscription."""
```

Comment:
Docstrings should not exceed 72 characters per line per PEP8. The first line is a header line with a single sentence.